### ISSUE TITLE
Clarification on how exactly 9g2 cutoffs by result work

### ISSUE THREAD

**Issue by jfly** [thewca/wca-regulations#426]
Emphasis mine:

> [9g2](https://www.worldcubeassociation.org/regulations/#9g2)) Whether a competitor proceeds to next phase of a Combined round, must be decided by ranking (best x competitors) or **by result (all competitors with a best result under x)** of the first phase.

Just to be clear about this, we have 3 "kinds" of results, depending on the event:

- FMC: Integer (number of turns) or DNF or DNS
- Multi bld: 2 integers + centiseconds or DNF or DNS
- Every other event: Centiseconds or DNF or DNS

A liberal interpretation of regulation 9g2 makes it sound like you could theoretically have a combined round of FMC mean of 3 where people must solve the first attempt in less than or equal to 30 moves in order to be allowed to participate in the second and third attempts.

Things get weird for multibld. [9f12c](https://www.worldcubeassociation.org/regulations/#9f12c) defines what a "result" is for mbld, and it says that "a greater difference [result] is better". This, combined with the fact that 9g2 says "all competitors with a best result **under** x" results in a funny conclusion: you'd have to do "worse" to advance to the next phase a combined round of mbld. (aplogies if I got this wrong, I read the relevant regulations *many* times before writing this).

Is there any reason to ever hold a combined round of FMC or a combined round of multibld? Supporting this will make our software significantly more complicated than I feel it is worth doing. Similarly with #425, in practice I bet these never happen, in part because cubecomps just doesn't support it.

I propose extending 9g2 to only allow combined rounds for events where results are measured in centiseconds.

**Comment by Claster** [comment]
Combined round in multibld was used in at least Euro 2016, 2014 and 2012, and will be used in WC 2017. The reason is probably to reduce the number of cubes organizers need to scramble, which becomes an issue at big competitions.

**Comment by SAuroux** [comment]
Hi Jeremy, you are right with "under" being generally wrong there. But
there are reasons to hold combined rounds for FMC and MBF: logistics.

Euros 2014 and 2016 had cutoffs (6points) for MBD for example, to reduce
the amount of staff work and logistics for the 2nd attempt compared to the
first attempt. This worked very well in practice.



2017-04-15 19:57 GMT+02:00 Jeremy Fleischman <notifications@github.com>:

**Comment by jfly** [comment]
Interesting. https://euro2016.cubing.net/events announced the following for multibld:

![image](https://cloud.githubusercontent.com/assets/277474/25065949/e3dffae6-21cd-11e7-9dd0-61400e98148e.png)

So, the cutoff was 6 points, which by 9g2 means that competitors "all competitors with a best result under 6 points" were eligible for phase 2 of the round. Oleg, I see you got 18 - 2 = 16 points on the first solve of the round, which means you should not have qualified for phase 2 of the round =)

**Comment by jfly** [comment]
Also worth noting that both Tobias Peter and Axel Brisse got 6 points on the first solve, but were allowed to compete in the second phase:

![image](https://cloud.githubusercontent.com/assets/277474/25066004/069bf840-21cf-11e7-9ff7-d8203063f5d9.png)

It sounds like multibld cutoffs are useful in practice, so when we update the regulation, it should be updated to be clear about "less than" vs "less than or equal to".

**Comment by Laura-O** [comment]
Agree with the comments regarding the logistics: MultiBLD attempt 1 at Euro 2016 was definitely one of my worst experiences as an organizer and attempt 2 was just smooth and easy... :-)

About the other issue: would it make sense to rephrase it like this: 
> 9g2) Whether a competitor proceeds to next phase of a Combined round, must be decided by ranking (best x competitors) or by result (*all competitors with a better result than x*) of the first phase.

*Better* is defined in [#9f12](https://www.worldcubeassociation.org/regulations/#9f12), so this might work?

**Comment by Claster** [comment]
I remember seeing cut-offs like "best of 2 / average of 5 (<2:00)", and 2:00 results are not accepted in this case.
Therefore I think it is rather organizers should specify whether the cutoff requires "strictly better" or "better or equal" result. E.g., "Multi-bld, best of 1 / best of 2 (>=6 points)".

**Comment by jfly** [comment]
> Better is defined in #9f12, so this might work?

That could work, but 9f12 talks about more than just multibld points. Do we want people to be able to announce cutoffs like "better than 6 points at 30 minutes"?

> Therefore I think it is rather organizers should specify whether the cutoff requires "strictly better" or "better or equal"

Ok, I don't care too much what the regulations say here, but we're only going to support one of these on the WCA website. It's just not worth the complexity for the WCA website and all downstream software to have to support both, when only one is needed.

**Comment by AlbertoPdRF** [comment]
> Therefore I think it is rather organizers should specify whether the cutoff requires "strictly better" or "better or equal" result. E.g., "Multi-bld, best of 1 / best of 2 (>=6 points)".

I don't agree here. The regulation clearly states "with a best result under x" or, with the change proposed in this thread, "with a better result than x". So the regulation requires strictly better.

Also, I like Laura's suggestion, as "better result" makes sense for all the events.

> That could work, but 9f12 talks about more than just multibld points. Do we want people to be able to announce cutoffs like "better than 6 points at 30 minutes"?

But all the other things that are said there are only applicable if there's a tie (I think), so that shouldn't be a problem. Only cutoffs by points should be set IMO.

**Comment by Claster** [comment]
Well, it doesn't matter if the cutoff is >=6 points or >5 points; or if it is <2:00 or <=1:59.99. However, if it written "6 points", it might lead to ambiguity, that's why I'd like organizers to specify ">" or ">=" explicitly. Explicit is better than implicit.

Regarding "6 points at 30 minutes", that's something not impossible under our current Regulations, and I can't come up with a beautiful fix that prohibits that. Even "6 points at 30 minutes and 8 solved cubes" is possible (which of course means that 7 out of 7 with the time 50 minutes is better than this cutoff). Though I don't think anything besides points cutoff was or will be ever used.

**Comment by jfly** [comment]
> But all the other things that are said there are only applicable if there's a tie

That's true, and an argument for saying cutoffs should always be "better than", not "better than or equal to".

> Explicit is better than implicit.

[`import this`](https://www.python.org/dev/peps/pep-0020/) =)

@Claster, I definitely agree that whenever someone writes down the cutoff, they should be explicit about it. The discussion here isn't about how you write it down, it's about what kinds of cutoffs are legal (and tangentially, which cutoffs our software will support). I'm telling you that the software is only going to do one kind of cutoff ("better" or "better or equal to"), not both, so it's up to the WRC how you want to handle that =)

> Regarding "6 points at 30 minutes", that's something not impossible under our current Regulations

I think if you read our regualtions very carefully, you'll see that it actually is impossible to say "6 points at 30 minutes" currently. Here's how I get that:

- [9g2](https://www.worldcubeassociation.org/regulations/#9g2) clearly talks about "results"
- This is the tricky part. If you read [9f12c](https://www.worldcubeassociation.org/regulations/#9f12c) carefully, you'll see it implicitly defines "result" for multibld as "number of puzzles solved minus the number of puzzles not solved". Just think very carefully about what the word "result" must be referring to in "If competitors achieve the same result..."

In other words, the only possible interpretation of 9f12c is that a competitor who achieved 8/12 in 34 minutes got exactly the same result as a competitor who achieved 8/12 in 36 minutes and exactly the same result as a competitor who got 4/4 in 22 minutes.

**Comment by Claster** [comment]
> I'm telling you that the software is only going to do one kind of cutoff ("better" or "better or equal to"), not both

As I said, those are equivalent anyway, so it doesn't matter much.
Can it be "strictly better" for all timed events and "better or equal" for MBLD and FMC?
If not, what terrible can happen if there is a misunderstanding between organizer's and software's understanding of strict and non-strict inequalities? E.g., an organizer wants to set a cutoff ">=6 points", enters 6 and it is interpreted as ">6 points"?
In other words, what do you need it for? :)

**Comment by AlbertoPdRF** [comment]
> Can it be "strictly better" for all timed events and "better or equal" for MBLD and FMC?

I think that this will only add confusion, so we better decide to use "strictly better" or "better or equal" for all events.

**Comment by pedrosino** [comment]
@jfly is right about "result" being only points for Multi BLD.

I think Laura's proposal fixes the issue, and the use of > or >= is another thing :)

**Comment by AlbertoPdRF** [comment]
As Laura's suggestion seems to fit perfectly, the only thing left to discuss is if we use _better than x_ or _better or equal to x_.
